tests/test_anonfile.py

```diff
@@ -38,73 +39,116 @@ def remove_file(file: str) -> None:
+    @patch('anonfile.requests.Session.post')
+    def test_upload(self, mocked_session_post):
+        """ Tests a mocked upload """
+
+        # Given
+        json_response = MockData.get_json_response("tests/upload.json")
+
+        mocked_session_post.side_effect = [json_response]
+
+        # Assert
         upload = self.anon.upload(self.test_file, progressbar=True, enable_logging=True)
         self.assertTrue(upload.status, msg="Expected 200 HTTP Error Code")
         self.assertTrue(all([upload.url.scheme, upload.url.netloc, upload.url.path]), msg="Invalid URL.")
 
-    def test_preview(self):
-        preview = self.anon.preview(self.test_small_file)
+    @patch('anonfile.requests.Session.get')
+    def test_preview(self, mocked_session_get):
+        """ Tests mocked preview API request """
+
+        # Given
+        json_response = MockData.get_json_response("tests/preview.json")
+        html_response = MockData.get_html_response("tests/preview.html")
+
+        mocked_session_get.side_effect = [json_response, html_response]
+
+        # Assert
+        preview = self.anon.preview(self.test_med_file)
         self.assertTrue(preview.status, msg="Error in status property.")
-        self.assertEqual(self.test_small_file, preview.url.geturl(), msg="Error in URL property.")
-        self.assertEqual("93k5x1ucu0", preview.id, msg="Error in ID property.")
-        self.assertEqual("test.txt", preview.file_path.name, msg="Error in name property.")
-        self.assertEqual(271, preview.size, msg="Error in size property.")
+        self.assertEqual(self.test_med_file, preview.url.geturl(), msg="Error in URL property.")
+        self.assertEqual("P0mev3tfz7", preview.id, msg="Error in ID property.")
+        self.assertEqual("topsecret.mp4", preview.file_path.name, msg="Error in name property.")
+        self.assertEqual(3537832, preview.size, msg="Error in size property.")
+
+    @patch('anonfile.requests.Session.get')
+    def test_download(self, mocked_session_get):
+        """ Tests mocked file download with a simulated stream """
+
+        # Given
+        json_response = MockData.get_json_response("tests/preview.json")
+        html_response = MockData.get_html_response("tests/preview.html")
+        file_response = MockData.get_file_response("tests/original_topsecret.mp4")
 
-    def test_download(self):
-        download = self.anon.download(self.test_small_file, progressbar=True, enable_logging=True)
+        mocked_session_get.side_effect = [json_response, html_response, file_response]
+
+        download = self.anon.download(self.test_med_file, progressbar=True, enable_logging=True)
         self.assertTrue(download.file_path.exists(), msg="Download not successful.")
-        self.assertEqual(download.file_path.name, self.test_file.name, msg="Different file in download path detected.")
+        self.assertEqual(download.file_path.name, "topsecret.mp4", msg="Different file in download path detected.")
```

Comment:
you should derive this name from `cls.test_file`, so that the test file name is only hard-coded in one place, makes it easier to rename the file afterwards if necessary